colcon_bundle/verb/bundle.py

```diff
@@ -283,3 +273,69 @@ def _get_jobs(self, args, installers, decorators):
 
             jobs[pkg.name] = job
         return jobs
+
+    def _check_package_dependency_update(self, bundle_base, decorators):
+
+        dependency_hash = {}
+
+        for decorator in decorators:
+            if not decorator.selected:
+                continue
+            pkg = decorator.descriptor
+            dependency_list = sorted(dependency.name for dependency in
+                                     pkg.dependencies['run'])
+            dependency_hash[pkg.name] = hashlib.sha256(
+                ' '.join(dependency_list).encode('utf-8')).hexdigest()
+
+        current_hash_string = json.dumps(dependency_hash, sort_keys=True)
+        logger.debug('Hash for current dependencies: '
+                     '{current_hash_string}'.format_map(locals()))
+
+        dependency_hash_path = os.path.join(
```

Comment:
These paths should only be defined once, can we merge the ones above and these? I've been using parameters but that causes large parameter lists, maybe there could be an object with all the paths? Up to you how you want to implement.